core/src/main/java/org/apache/druid/java/util/common/UnsafeUtil.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.common;
+
+import java.lang.reflect.Field;
+
+public class UnsafeUtil
```

Comment:
Please add a Javadoc comment to this class, and/or methods. At least it should mention that this class must be accessed only in after checking that Java 8 is used via `JvmUtils`.

Reply:
This code works in Java 9 and above as well, we use it in the Java 9 version of the ByteBufferUtil code. Reflection is required because those classes are not available at compile time anymore.

Reply:
Please move this comment to code.